ui/static/ui/js/utils.jsx

```diff
@@ -1,12 +1,75 @@
-define("utils", ["jquery", "lodash", "react", "select2"],
-  function ($, _, React) {
+define("utils", ["jquery", "lodash", "react", "react_infinite", "select2"],
+  function ($, _, React, Infinite) {
   'use strict';
 
   /**
-   * Get a collection from the REST API
+   * A list which connects to our collection APIs and grows as the user
+   * scrolls down.
+   */
+  var InfiniteList = React.createClass({
+    getInitialState: function() {
+      return {
+        elements: [],
+        isInfiniteLoading: false,
+        next: this.props.url
+      };
+    },
+    componentDidMount: function() {
+      this.loadPage();
+    },
+    loadPage: function() {
+      var thiz = this;
+
+      if (this.state.next === null) {
+        this.setState({
+          isInfiniteLoading: false
+        });
+        return;
+      }
+
+      $.get(this.state.next).then(function (results) {
+        thiz.setState({
+          elements: thiz.state.elements.concat(results.results),
+          next: results.next,
+          isInfiniteLoading: false
+        });
+      }).fail(function(err) {
+        thiz.setState({
+          isInfiniteLoading: false
+        });
+        thiz.props.onError(err);
+      });
+    },
+    handleInfiniteLoad: function () {
+      this.setState({
+        isInfiniteLoading: true
+      });
+      this.loadPage();
+    },
+    render: function () {
+      var spinner = <div className="infinite-item-list">Loading...</div>;
```

Comment:
This was pulled from their example code, I didn't do much thinking on how this would look. I have no argument switching to a spinner it may be better to do it in a separate PR